docs/concepts/what-is-a-workflow-execution.md

```diff
@@ -126,3 +126,19 @@ A Workflow Execution Timeout can be used to limit the duration of Workflow Execu
 You can use the [Continue-As-New](/concepts/what-is-continue-as-new) feature to close the current Workflow Execution and create a new Workflow Execution in a single atomic operation.
 The Workflow Execution spawned from Continue-As-New has the same Workflow Id, a new Run Id, and a fresh Event History and is passed all the appropriate parameters.
 For example, it may be reasonable to use Continue-As-New once per day for a long-running Workflow Execution that is generating a large Event History.
+
+### Limits
+
+When executing Commands, Temporal allocates resources that get released once a Workflow Task has completed or terminated.
+
+For example, each execution of `ScheduleActivityCommand` adds an Activity to the Task Queue, along with the resources needed to execute it.
+However, scheduling too many Activities can take resources away from what's currently running.
+
+To avoid this scenario, Temporal fails Workflow Task Executions that would cause the Workflow to surpass 50,000 pending Activities.
+Similar constraints are enforced for `SignalExternalWorkflowExecution`, `RequestCancelExternalWorkflowExecution`, and `StartChildWorkflowExecution` Commands.
+
```

Comment:
This is not accurate.

The correct one would be something like this:
Each pending activity (activity scheduled but not completed yet) will generate metadata entry in workflow mutable state. Too many such entries will result in mutable state become too big. At large scale this will result in persistence become unstable. To protect the system, Temporal enforce max pending activity limit. If incoming command would lead to pending activity exceed that limit, temporal server will reject the command causing the workflow task to fail.  By default the limit is 50000 which is essentially no limit because it is the same as event count limit. But we recommend setting some reasonable limit (for example 1000) to protect the system.